File: tasks/1_Two_Sum.go
package main

// https://leetcode.com/problems/two-sum/
func twoSum(nums []int, target int) []int {
	cache := make(map[int]int, len(nums))
	for pos, num := range nums {
		if pos2, ok := cache[target-num]; ok {
			return []int{pos, pos2}
		}

		cache[num] = pos
	}

	return nil
}

File: tasks/1_Two_Sum.py
# https://leetcode.com/problems/two-sum/
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = {}

        for pos, num in enumerate(nums):
            if (pos2 := cache.get(target - num)) is not None:
                return [pos2, pos]

            cache[num] = pos
